Added OCR typo transformation

goyalpramod opened the pull request:
## Description

<!-- Add a more detailed description of the changes if needed. -->
Added OCR typo transformation. I have only added replacement and deletion as they seem to be the most common type of error in OCR as mentioned [here](https://aclanthology.org/W96-0108.pdf)

## Related Issue

<!-- If your PR refers to a related issue, link it here. -->
#1565 

## Type of Change

<!-- Mark with an `x` all the checkboxes that apply (like `[x]`) -->

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

<!-- Mark with an `x` all the checkboxes that apply (like `[x]`) -->

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

andreybavt commented on giskard/scanner/robustness/text_transformations.py:
```diff
@@ -122,7 +122,49 @@ def _random_key_typo(self, char):
+
+    def make_perturbation(self, x):
+        # Skip if the text is too short
+        if len(x) < self.min_length:
```
there should also be a `None` check, otherwise it'll fail at `len(None)`

andreybavt commented on tests/scan/test_text_transformations.py:
```diff
@@ -247,3 +247,11 @@ def test_typo_transformation():
+    t = TextFromOCRTypoTransformation(column="text", rng_seed=1)
+    p = t.make_perturbation("If one doesn't know his mistakes, he won't want to correct them.")
+
+    assert p == "1f 0ne d0ezn't kn0uu his nnistakes, he w0n't want to correct tbenn."
```
The test doesn't pass for me:

```
Expected :"1f 0ne d0ezn't kn0uu his nnistakes, he w0n't want to correct tbenn."
Actual   :"If one doesn't know his mi5takes, he won't want to corrct them."

```

andreybavt replied:
Could you also add a test for the `len(x) < self.min_length` condition?

goyalpramod replied:
I will check it and get it done.

goyalpramod commented:
@andreybavt I apologize for the late fix, I had gone on a new year vacation. Happy New Year. I have fixed the error (I was using another random seed in testing). And added the things you recommend. Kindly let me know if I should do anything else, Thank you.

kevinmessiaen commented:
Thank

> @andreybavt I apologize for the late fix, I had gone on a new year vacation. Happy New Year. I have fixed the error (I was using another random seed in testing). And added the things you recommend. Kindly let me know if I should do anything else, Thank you.

Thanks @goyalpramod and happy new year! No worries about it, hope enjoyed your vacation!
